vinhloc30796/creativecontact-app
# feat(i18n)!: add internationalization for upload & artwork components

## Description

This PR implements internationalization (i18n) support for the upload and artwork components, enhancing the application's multilingual capabilities.

## Key Changes

1. Add translations for English and Vietnamese
2. Implement i18n in ArtworkCreditInfoStep, ArtworkInfoStep, ContactInfoStep, ProfessionalInfoStep, and MediaUpload components
3. Update UploadPageClient and UploadConfirmedContent to use translations
4. Modify RootLayout and Providers to handle language dynamically
5. Add SearchParamsProvider for language management
6. Create new Loading component for Suspense fallback

## Breaking Changes

- The RootLayout and Providers components no longer accept a 'lang' prop. Language is now determined from search params.
- User-facing text is now managed through i18n, requiring updates to text content workflow

## Related Issues

- Addresses WEB-99

## Testing Instructions

1. Run the application and verify that language switching works correctly
2. Test the upload and artwork components in different languages
3. Verify that the Loading component appears during language transitions

## Additional Notes

This PR builds upon the previous internationalization work, extending it to more components and improving the overall i18n implementation. Reviewers should pay special attention to the language switching mechanism and the correct display of translated content in different parts of the application.
